src/java.base/share/classes/java/io/SequenceInputStream.java

```diff
@@ -80,7 +79,18 @@ public SequenceInputStream(Enumeration<? extends InputStream> e) {
      * @param   s2   the second input stream to read.
      */
     public SequenceInputStream(InputStream s1, InputStream s2) {
-        this(Collections.enumeration(Arrays.asList(s1, s2)));
+        this(List.of(s1, s2).iterator());
```

Comment:
This changes the behaviour for null streams. This line will throw NPE. The previous code throwed NPE in `peekNextStream`, only if the stream was used.

Have your make sure that it is okay to change behaviour?

Reply:
good catch, switching back to Arrays.asList then.